WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.kt

```diff
@@ -309,4 +336,17 @@ class PostsListActivity : AppCompatActivity(),
     override fun onDismissByOutsideTouch(instanceTag: String) {
         viewModel.onDismissByOutsideTouchForBasicDialog(instanceTag)
     }
+
+    // Menu ViewLayoutType handling
+
+    private fun updateMenuIconForViewLayoutType(viewLayoutType: ViewLayoutType) {
```

Comment:
This is the only thing I might disagree with; I think this logic is already as much in the ViewModel as can be, because the actual icon reference should not be in the ViewModel layer at all. The view model should have no knowledge of the inner working of Android systems so that it can still be tested in a pure unit test environment rather than an Android Test environment. Unless I'm missing something, attempting to move this into the view model would required the view model having knowledge about the actual icons?

Reply:
>The view model should have no knowledge of the inner working of Android systems so that it can still be tested in a pure unit test environment rather than an Android Test environment.

I can't agree more with this sentence.

> attempting to move this into the view model would required the view model having knowledge about the actual icons?

I don't believe that's the case. The id of an icon is just a simple Integer from a generated java file - `R.java`. We use both string and drawable resources (their ids to be more specific) in our view models and unit tests work as expected.

Reply:
Ah interesting, I never use R files in unit tests since they seemed so Android specific. OK, sure, I'll change this, less untestable logic is never a bad thing!